libs/dh/admin/data-access-api/src/lib/dh-admin-user-roles.store.ts

```diff
@@ -85,13 +117,51 @@ export class DhAdminUserRolesStore extends ComponentStore<DhUserManagementState>
     )
   );
 
+  readonly assignRoles = this.effect(
+    (
+      trigger$: Observable<{ userId: string; updateUserRoles: UpdateUserRoles }>
+    ) => {
+      return trigger$.pipe(
+        concatMap(({ userId, updateUserRoles }) =>
```

Comment:
suggestion: `concatMap` can probably be replaced by `exhaustMap` here. So the logic afterwards won't execute again until the inner Observables complete (in case `assignRoles` is called multiple times for whatever reason).

Reply:
Fixed in https://github.com/Energinet-DataHub/greenforce-frontend/pull/1374/commits/295086a7a9e7b5814880aba180eb59947be5ffde